pkg/segment/structs/evaluationstructs.go

```diff
@@ -1876,6 +1913,16 @@ func (self *ValueExpr) EvaluateValueExprAsString(fieldToValue map[string]utils.C
 	return str, nil
 }
 
+func isNumber(str string) bool {
+	_, err := strconv.ParseFloat(str, 64)
+	return err == nil
+}
```

Comment:
I think it would be better to move this and the below functions into `utils` package, specifically to `pkg/segment/utils/segconsts.go` and make them public. This is the place where the above types are also defined. 

In that way in future we might use them at some other places.